go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -231,20 +232,16 @@ func (oc *Controller) syncEgressFirewall(egressFirwalls []interface{}) {
 
 func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.EgressFirewall, txn *util.NBTxn) error {
 	klog.Infof("Adding egressFirewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
-	nsInfo, err := oc.waitForNamespaceLocked(egressFirewall.Namespace)
-	if err != nil {
-		return fmt.Errorf("failed to wait for namespace %s event (%v)",
-			egressFirewall.Namespace, err)
-	}
-	defer nsInfo.Unlock()
-
-	if nsInfo.egressFirewall != nil {
+	oc.egressFirewallsMutex.Lock()
+	defer oc.egressFirewallsMutex.Unlock()
```

Comment:
you are going to hold this lock while addEgressFirewallRules runs and creates all the OVN things right? This is essentially making egressFirewallHandler only of capable of handling a single firewall until each is completely configured in OVN. In the old logic you would have been able to handle multiple at a time. Is this a concern?

Reply:
If it is, i would suggest using a sync map to hold all of your egressFirewalls per namespace, then adding a Mutex to the egressFirewall struct. Attempt to only lock the egressFirewall that you care about during operations, rather than locking up the entire map.

Reply:
Yes, i have a similar concern. The addition of EgressFirewall objects has now been serialized with this PR.

Reply:
I think I addressed this with the config map